2004-05-04  Jakub Jelinek  <jakub@redhat.com>

	* config/tc-s390.h (md_do_align): Pass (1 << n) - 1 instead of 15
	as max_chars to frag_var.

--- gas/config/tc-s390.h.jj	2003-07-10 15:03:13.000000000 +0200
+++ gas/config/tc-s390.h	2004-05-04 16:42:12.724802992 +0200
@@ -80,7 +80,7 @@ extern int target_big_endian;
 if ((n) && !need_pass_2 && (fill == 0) &&                               \
     (bfd_get_section_flags (stdoutput, now_seg) & SEC_CODE) != 0) {     \
   char *p;                                                              \
-  p = frag_var (rs_align_code, 15, 1, (relax_substateT) max,            \
+  p = frag_var (rs_align_code, n > 4 ? ((1 << (n)) - 1) : 15, 1, (relax_substateT) max,\
                 (symbolS *) 0, (offsetT) (n), (char *) 0);              \
   *p = 0x07;                                                            \
   goto around;                                                          \
